File: app/src/main/java/com/pengxh/audodingding/BaseApplication.java
package com.pengxh.audodingding;

import android.app.Application;

import com.pengxh.app.multilib.utils.SaveKeyValues;
import com.pengxh.app.multilib.widget.EasyToast;

/**
 * @author: Pengxh
 * @email: 290677893@qq.com
 * @description: TODO
 * @date: 2019/12/25 13:19
 */
public class BaseApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        EasyToast.init(this);
        SaveKeyValues.initSharedPreferences(this);
    }
}

File: app/src/main/java/com/pengxh/autodingding/BaseApplication.kt
package com.pengxh.autodingding

import android.app.Application
import android.database.sqlite.SQLiteDatabase
import com.pengxh.autodingding.greendao.DaoMaster
import com.pengxh.autodingding.greendao.DaoSession
import com.pengxh.autodingding.utils.DingDingUtil
import com.pengxh.kt.lite.utils.SaveKeyValues
import com.tencent.bugly.crashreport.CrashReport
import kotlin.properties.Delegates

/**
 * @author: Pengxh
 * @email: 290677893@qq.com
 * @date: 2019/12/25 13:19
 */
class BaseApplication : Application() {

    companion object {
        private var application: BaseApplication by Delegates.notNull()

        fun get() = application
    }

    lateinit var daoSession: DaoSession

    override fun onCreate() {
        super.onCreate()
        application = this
        DingDingUtil.init()
        SaveKeyValues.initSharedPreferences(this)
        CrashReport.initCrashReport(this, "ce38195468", false)
        initDataBase()
    }

    private fun initDataBase() {
        val helper = DaoMaster.DevOpenHelper(this, "DingRecord.db")
        val db: SQLiteDatabase = helper.writableDatabase
        val daoMaster = DaoMaster(db)
        daoSession = daoMaster.newSession()
    }
}